doc/osi.sgml

```diff
@@ -0,0 +1,181 @@
+
+Example for converting an executable "hello" that was built for the default
+start address &dollar;0200 to an uploadable file "hello.c1p":
+
+<tscreen>
+srec_cat hello -binary -offset 0x200 -o hello.c1p -Ohio_Scientific -execution-start-address=0x200
+</tscreen>
+
+<sect>Memory layout<p>
+
+By default programs compiled for the osic1p target are configured for 32 kB RAM.
+The RAM size can be configured via the symbol __HIMEM__.
+
+Special locations:
+
+<descrip>
+  <tag/Program start address/
+  The default start address is &dollar;0200. The start address is configurable
+  via the compiler option --start-addr.
+
+  <tag/Stack/
+  The C runtime stack is located at the top of RAM and growing downwards.
+  The size is configurable via the symbol __STACKSIZE__. The default
+  stack size is &dollar;0400.
+
+  <tag/Heap/
+  The C heap is located at the end of the program and grows towards the C
+  runtime stack.
+
+  <tag/Video RAM/
+  The 1 kB video RAM is located at &dollar;D000. On the monitor only a subset
+  of the available video RAM is visible. The address of the upper left corner
+  of the visible area is &dollar;D085 and corresponds to conio cursor
+  position (0, 0).
+
+</descrip><p>
+
+Example for building a program with start address &dollar;0300, stack size
+&dollar;0200 and RAM size &dollar;2000:
+
+<tscreen>
+cl65 --start-addr 0x300 -Wl -D,__HIMEM__=$2000,-D,__STACKSIZE__=$0200 -t osic1p hello.c
+</tscreen>
+
+<sect>Linker configurations<p>
+
+The ld65 linker comes with a default config file "osic1p.cfg" for the Ohio Scientific
+Challenger 1P, which is implicitly used via <tt/-t osic1p/. The
+osic1p package comes with additional secondary linker config files, which are
+used via <tt/-t osic1p -C &lt;configfile&gt;/.
+
+<sect1>default config file (<tt/osic1p.cfg/)<p>
+
+The default configuration is tailored to C programs.
+
+<sect1><tt/osic1p-asm.cfg/<p>
+
+This configuration is made for assembler programmers who don't need a special
+setup.
+
+To use this config file, assemble with <tt/-t osic1p/ and link with
+<tt/-C osic1p-asm.cfg/. The former will make sure that correct runtime library
+is used, while the latter supplies the actual config. When using <tt/cl65/,
+use both command line options.
+
+Sample command line for <tt/cl65/:
+
+<tscreen><verb>
+cl65 -o program -t osic1p -C osic1p-asm.cfg source.s
+</verb></tscreen>
+
+<sect>Platform-specific header files<p>
+
+Programs containing Ohio Scientific-specific code may use the <tt/osic1p.h/ 
+header file. 
+
+<sect1>Ohio Scientific-specific functions<p>
+
+There are currently no special Ohio Scientific functions.
+
+<sect1>Hardware access<p>
+
+There is no specific support for direct hardware access.
+
+<sect>Loadable drivers<p>
+
+There are no loadable drivers available.
+
+<sect>Limitations<p>
+
+<sect1>conio implementation<p>
+
+The conio implementation is complete except for the kbhit() function. A
+call to cgetc() always blocks until a character is entered.
```

Comment:
Isn't the statement

> A call to cgetc() always blocks until a character is entered.

generally true for all implementations? Or do you want to state that a key typed before entering cgetc() can't be fetched. In the latter case you should probably state that (even more) explicitly.